### PR TITLE
almost completed assessment

### PR DESCRIPTION
**Pull request by JPD99**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [x] JavaScript Coding Challenges
- [x] Ruby Coding Challenges
- [x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

**Comment by JPD99** [comment]
a few of the interview questions aren't finished; i should be able to get them done tonight.

**Comment by WalkerTexas-coder** [comment]
Okay I'll be looking for those interview questions tomorrow morning!

**Comment by JPD99** [comment]
Just sent it in, I hope the changes are sufficient!

File: interview-questions.md [hunk 1]
```diff
@@ -8,44 +8,127 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+          :cranberry_juice => 2}
+juices[:apple_juice] = 4 
+OR
+juices = {apple_juice: 4 ,
+          cranberry_juice: 2}
+juices[:apple_juice] = 4 
+
+The former and latter have different notations for their key value pairs, but note that they way to call back a specific key value pair is identical. 
+
+Additional intresting point is that Hashes can store other referential data types: arrays, objects, and even more hashes. The thing is, is that a hash within a array within a hash may be very diffcult to read.
```
Comment (SunkissedQueen) [inline interview-questions.md:28]: nice

File: interview-questions.md [hunk 2]
```diff
@@ -8,44 +8,127 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-Researched answer:
 
 2. What is a gem?
 
-Your answer:
+Your answer: Gems are library components of Rails, very similiar to React's library. They both offer seemless paths for integrating other people's components into a new project. 
 
 Researched answer:
+A gem is an indivdual package from a Ruby library. It's almost like installing an app, as a gem has a name, and version number associated to it. Gems are essentially the same thing as a gem, but its plural. Gems are a collection of libraries or applications put into one package that can be installed with one command line: use gem install [gem_name]. I personally have not installed a gem package before, but I'm sure I have used it before as a testing suite for Ruby code.
```
Comment (SunkissedQueen) [inline interview-questions.md:35]: We have install ruby gems. Look at rspec

File: interview-questions.md [hunk 3]
```diff
@@ -8,44 +8,127 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
 
 Researched answer:
+Ruby on Rails is a framework application written for Ruby, plus some other integral langauges (javascript, css, and html); if a full stack application is made from Ruby on Rails, that same application can have both front-end and back-end technologies. The entire structure of Ruby on Rails uses the MVC Architecture: Model, View, and Controller to operate the user interface aspect; model is how the data is shaped, and can be used to relate other data; view is what the user sees, controller is the brain of the application, manages interaction between user and database. Ruby on Rails has a ton of functionality that is created automatically! The only drawback of this framework is that it is very, very picky with syntax, improper spelling + incorrect naming conventions will cause a lot of issues; this trade off is refered to as convention from configuration.
+
```
Comment (SunkissedQueen) [inline interview-questions.md:44]: like that drawback content

File: interview-questions.md [hunk 4]
```diff
@@ -8,44 +8,127 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+    $ rails db:migrate
+    $ irb
+Ruby Console:
+    Team.create team_name:'Cowboys' , city:'Dallas'
+    cowboys = Team.find 1
+    cowboys.players.create person_name:'Bobby Joe' 
+        <!-- Might notice why we didn't fill in the user_id parameter... this way automatically asigns the primary key of the value in Team into the team_id (aka the foregin key)  -->
+
+Also I did manage to find other forms of databases. There seems to be an entire encyclopedia worth of database types! To name a few there is: Operational Database, Distributed Database, Cloud Database, End User Database, Centralized Database, ... and so on and so forth. 
+
```
Comment (SunkissedQueen) [inline interview-questions.md:69]: great approach

File: interview-questions.md [hunk 5]
```diff
@@ -8,44 +8,127 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+        id: 2. ,
+        person_name: 'Bobby Joe'
+        team_id: 1,
+        created_at: Mon, 6 Sept 2022 23:12:42.703053000 UTC +00:00,
+        updated_at: Mon, 6 Sept 2022 23:12:42.703053000 UTC +00:00]
+
+We see the automated id column is created automatically and registers a new number for every record created. 
+We also see that in the Player database the foregin key, team_id, is linked to primary key 1, theeby establishing a relational connection across databases
+
+
```
Comment (SunkissedQueen) [inline interview-questions.md:116]: bravo

**Review by SunkissedQueen — APPROVED** [review]
Jereme, based on the conversation between you and Austin, I believe he has given you feedback on everything else. If not, then reach out to me with any questions on the code and ruby challenges. This approach that you are taking on the interview questions is fantastic. You are showing that you understand the material not only with the definition but with appropriate examples. Remember that during your interviews. Nicely done.

**Comment by JPD99** [comment]
thank you!

**Pull request merged**